### PR TITLE
2.x Remove twig/cache-extension, add support for twig/cache-extra

### PR DESCRIPTION
**Pull request by gchtr**
**Ticket**: #2407

## Issue

Following the discussions in #2486, I think I may have a solution for replacing the `twig/cache-extension` package.

## Solution

This pull request:

- Removes the `twig/cache-extension` package.
- Removes all interfaces that it provided, but keeps the classes that used them.
- Adds the `twig/cache-extra` in composer.json’s `suggest` field.
- Adds tests to test the support for the `twig/cache-extra` package.
- Updates Timber’s Cache Key Generator to not use any of the characters that are forbidden to use as cache keys with the `twig/cache-extra` package.
- Adds documentation about using the `twig/cache-extra` package.
- Adds the relevant information in the Upgrade Guide.

## Impact

- Breaks compatibility for people who use the `{% cache %}` tag in Twig.

## Usage Changes

The `{% cache %}` works a little different. You don’t provide it with a variable that will be cached anymore. It will cache the contents of the tag based on what you provide as the cache key.

Timber’s Cache Key Generator still works with it and could be a helpful tool. For now, I added some documentation for how to use it.

## Considerations

- Is it bad to remove the interfaces?
- In the future, PSR-6 and PSR-16 might be helpful.

## Testing

Yes, I added some tests, but they only test that using the `twig/cache-extra` produces any errors and not if the contents are actually cached.

### CODE REVIEW

**Review by nlemoine — CHANGES_REQUESTED** [review]
This is a great compromise for dealing with this cache extension issue! Good work @gchtr! 👍

File: composer.json [hunk 1]
```diff
@@ -30,8 +30,7 @@
   "require": {
     "php": "^7.4|^8.0",
     "twig/twig": "^2.12|^3.0",
-    "composer/installers": "^2.0",
-    "twig/cache-extension": "^1.5.0"
+    "composer/installers": "^2.0"
```
[the remark this answers is not in the document]
Comment (gchtr) [reply composer.json:33]: Updated in #2550.

**Comment by gchtr** [comment]
@jarednova @nlemoine Just need another pair of eyes to look over this and approve this pull request, then we can merge this in.

**Review by jarednova — APPROVED** [review]
Very helpful upgrade guide and docs as always. Great call to utilize the `suggest` part of Composer — just noticed that nice detail

**Comment by jarednova** [comment]
Looks like it just needs @nlemoine 🙏  for his [previously requested change](https://github.com/timber/timber/pull/2546#pullrequestreview-894393017)

**Review by nlemoine — APPROVED** [review]

**Pull request merged**